### PR TITLE
Website Cleanup Part 6 - Proposals (Part 2)

### PR DESCRIPTION
**Pull request by n-boshnakov**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR proofreads the Proposals section of the documentation. This is Part 2 of 2.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc developer
NONE
```

### CODE REVIEW

File: docs/proposals/20-ha-control-planes.md [hunk 1]
```diff
@@ -713,52 +712,52 @@ Cross zonal traffic rates for some of the providers are:
...
+> **Disclaimer:** [Topology Aware Hints](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/) won't improve network traffic if the seed has worker nodes in more than three zones and the Kube Apiserver is scaled beyond three replicas at the same time. In this case, Kube Apiserver replicas run in zones which don't have an etcd and thus cross zone traffic is inevitable.
 
 _During evaluation of this feature some caveats were discovered:_
 
-For each cluster, gardener provides a capability to create one or more `Worker Pool/Group`. Each worker pool can span across one or more availability zones. For a combination of each worker pool and zone there will be a corresponding `MachineDeployment` which will also map 1:1 to a `Node Group` which is understood by cluster-autoscaler.
+For each cluster, Гardener provides a capability to create one or more `Worker Pool/Group`. Each worker pool can span across one or more availability zones. For a combination of each worker pool and zone there will be a corresponding `MachineDeployment` which will also map 1:1 to a `Node Group`, which is understood by cluster-autoscaler.
```
Comment (ialidzhikov) [inline docs/proposals/20-ha-control-planes.md]:
```suggestion
For each cluster, Gardener provides a capability to create one or more `Worker Pool/Group`. Each worker pool can span across one or more availability zones. For a combination of each worker pool and zone there will be a corresponding `MachineDeployment` which will also map 1:1 to a `Node Group`, which is understood by cluster-autoscaler.
```

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

**Review by ialidzhikov — COMMENTED** [review]
Why we keep updating the GEPs? Because when you navigate to https://gardener.cloud/docs/, they are part of the Gardener docs? Why they have to be part of the user facing docs? When I navigate to https://docs.k8s.io I don't see any KEPs. I would rather suggest excluding them from the gardener.cloud/docs and stop updating them with wording improvements or typo fixes.

**Comment by rfranzke** [comment]
I agree with @ialidzhikov. We can merge this PR now since work was invested already, though.

**Comment by ialidzhikov** [comment]
I would like to first see change that excludes the GEPs from https://gardener.cloud/docs.

**Comment by rfranzke** [comment]
> I would like to first see change that excludes the GEPs from https://gardener.cloud/docs.

Is this related to this PR in any way?

**Comment by n-boshnakov** [comment]
I have created a PR that removes the Proposals section from the public website, as per @ialidzhikov's request.

I'm open to any further suggestions.

File: docs/proposals/20-ha-control-planes.md [hunk 2]
```diff
@@ -95,54 +95,54 @@ Each consumer therefore needs to decide on the degree of failure isolation that
 
 ## Non-Goals
 
-- Setting up a high available Gardener service.
+- Setting up a high availability Gardener service.
```
Comment (ialidzhikov) [inline docs/proposals/20-ha-control-planes.md]: Isn't more correct to say the following?

```suggestion
- Setting up a highly available Gardener service.
```

**Review by ialidzhikov — COMMENTED** [review]
/lgtm

File: docs/proposals/03-networking-extensibility.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ Finally, every provider is different, and thus the network always needs to adapt
 
 ## Gardener Network Extension
 
-The goal of the Gardener Network Extensions is to support different network plugin, therefore, the specification for the network resource won't be fixed and will be customized based on the underlying network plugin. To do so, a `NetworkConfig` field in the spec will be provided where each plugin will be defined. Below is an example for deploying Calico as the cluster network plugin.
+The goal of the Gardener Network Extensions is to support different network plugin, therefore the specification for the network resource won't be fixed and will be customized based on the underlying network plugin. To do so, a `NetworkConfig` field in the spec will be provided where each plugin will be defined. Below is an example for deploying Calico as the cluster network plugin.
```
Comment (acumino) [inline docs/proposals/03-networking-extensibility.md]:
```suggestion
The goal of the Gardener Network Extensions is to support different network plugins, therefore the specification for the network resource won't be fixed and will be customized based on the underlying network plugin. To do so, a `NetworkConfig` field in the spec will be provided where each plugin will be defined. Below is an example of deploying Calico as the cluster network plugin.
```

File: docs/proposals/10-shoot-additional-container-runtimes.md [hunk 4]
```diff
@@ -97,7 +97,7 @@ Each extension will need to address the following concerns:
         2. Apply the relevant RuntimeClass to the cluster.
 
     2. The installation above should be done by a new kind of extension: ContainerRuntime resource. For each container runtime type (Kata-container/gvisor), a dedicate extension controller will be created.
-        1. A label for each container runtime support will be added to every node that belongs to the worker pool. This should be done similar to the way labels created today for each node, through kubelet execution parameters (`_kubelet.flags: --node-labels`). When creating the OperatingSystemConfig (original) for the worker, each container runtime support should be mapped to a label on the node.
+        1. A label for each container runtime support will be added to every node that belongs to the worker pool. This should be done similarly to the way labels created today for each node, through kubelet execution parameters (`_kubelet.flags: --node-labels`). When creating the OperatingSystemConfig (original) for the worker, each container runtime support should be mapped to a label on the node.
```
Comment (acumino) [inline docs/proposals/10-shoot-additional-container-runtimes.md]:
```suggestion
        1. A label for each container runtime support will be added to every node that belongs to the worker pool. This should be done similarly to the way labels are created today for each node, through kubelet execution parameters (`_kubelet.flags: --node-labels`). When creating the OperatingSystemConfig (original) for the worker, each container runtime support should be mapped to a label on the node.
```

File: docs/proposals/14-reversed-cluster-vpn.md [hunk 5]
```diff
@@ -140,19 +140,19 @@ The restriction regarding the `192.168.123.0/24` network range in the current VP
 
 ### Performance and Scalability
 
-We do expect performance and throughput to be slightly lower compared to the existing solution. This is because the OpenVPN server acts as an additional hop and must decrypt and re-encrypt traffic that passes through. As there are no low latency, or high thoughput requirements for this connection we do not assume this to be an issue.
+We do expect performance and throughput to be slightly lower compared to the existing solution. This is because the OpenVPN server acts as an additional hop and must decrypt and re-encrypt traffic that passes through. As there are no low latency or high thoughput requirements for this connection, we do not assume this to be an issue.
 
 ### Availability and Failure Scenarios
 
-This solution re-uses multiple instances of the envoy component used for the kube-apiserver endpoints. We assume that the availability for kube-apiservers is good enough for the cluster VPN as well.
+This solution reuses multiple instances of the envoy component used for the kube-apiserver endpoints. We assume that the availability for kube-apiservers is good enough for the cluster VPN as well.
```
Comment (acumino) [inline docs/proposals/14-reversed-cluster-vpn.md]:
```suggestion
This solution reuses multiple instances of the envoy component used for the kube-apiserver endpoints. We assume that the availability of kube-apiservers is good enough for the cluster VPN as well.
```

File: docs/proposals/14-reversed-cluster-vpn.md [hunk 6]
```diff
@@ -162,46 +162,46 @@ The change in the VPN solution will potentially open up new attack vectors. We w
...
+We achieve this by tunneling the open vpn connection through a WireGuard tunnel, which is established from the shoot to the seed (note that WireGuard uses UDP as protocol). Independent of that we can also use UDP for the OpenVPN connection, but we can also stay with TCP as it was before. While this might look like a big change, it only introduces minor changes to the existing solution, but let's look at the details. In essence, the OpenVPN connection does not require a public endpoint in the shoot cluster but it uses the internal endpoint provided by the WireGuard tunnel.
 
-This is roughly depcited in this diagram. Note, that the `vpn-seed` and `vpn-shoot` containers only require very little changes and are fully backwards compatible.
+This is roughly depcited in this diagram. Note that the `vpn-seed` and `vpn-shoot` containers only require very little changes and are fully backwards compatible.
```
Comment (acumino) [inline docs/proposals/14-reversed-cluster-vpn.md]:
```suggestion
This is roughly depicted in this diagram. Note that the `vpn-seed` and `vpn-shoot` containers only require very little changes and are fully backwards compatible.
```

File: docs/proposals/14-reversed-cluster-vpn.md [hunk 7]
```diff
@@ -162,46 +162,46 @@ The change in the VPN solution will potentially open up new attack vectors. We w
...
 
-While it is possible to configure a WireGuard device using DNS names only IP addresses can be stored in Linux Kernel data structures. A change of a load balancer IP address can therefore not be mitigated on that level. As WireGuard dynamically adapts endpoint IP addresses a change in load banlancer IPs is mitigated in most but not all cases. This is why a re-advertiser is required for public cloud providers such as AWS.
+While it is possible to configure a WireGuard device using DNS names, only IP addresses can be stored in Linux Kernel data structures. A change of a load balancer IP address can therefore not be mitigated on that level. As WireGuard dynamically adapts endpoint IP addresses, a change in load banlancer IPs is mitigated in most but not all cases. This is why a re-advertiser is required for public cloud providers such as AWS.
```
Comment (acumino) [inline docs/proposals/14-reversed-cluster-vpn.md]:
```suggestion
While it is possible to configure a WireGuard device using DNS names, only IP addresses can be stored in Linux Kernel data structures. A change of a load balancer IP address can therefore not be mitigated on that level. As WireGuard dynamically adapts endpoint IP addresses, a change in load balancer IPs is mitigated in most but not all cases. This is why a re-advertiser is required for public cloud providers such as AWS.
```

File: docs/proposals/18-shoot-CA-rotation.md [hunk 8]
```diff
@@ -71,30 +71,30 @@ In each of those cases we currently need to basically manually recreate and repl
...
-  - shoot reconciliation needs to be idemptotent to CA rotation phase, i.e. if a usual reconciliation or maintenance operation is triggered in between, no new CAs are generated or similar things that would interfere with the CA rotation sequence
+  - Annotation increases the generation
+  - Value for triggering first phase: `start-ca-rotation`
+  - Value for triggering the second phase: `complete-ca-rotation`
+  - `gardener-apiserver` performs the needful validation: a user can't trigger another rotation if one is already in progress, a user can't trigger `complete-ca-rotation` if first phase has not been compeleted, etc.
+- The annotation triggers a usual shoot reconciliation (just like a kubeconfig or SSH key rotation).
+- The `gardenlet` begins the CA rotation sequence by setting the new status section `.status.credentials.caRotation` (probably in `updateShootStatusOperationStart`) and removes the annotation afterwards.
+  - Shoot reconciliation needs to be idemptotent to CA rotation phase, i.e., if a usual reconciliation or maintenance operation is triggered in between, no new CAs are generated or similar things that would interfere with the CA rotation sequence.
```
Comment (acumino) [inline docs/proposals/18-shoot-CA-rotation.md]:
```suggestion
  - Shoot reconciliation needs to be idempotent to CA rotation phase, i.e., if a usual reconciliation or maintenance operation is triggered in between, no new CAs are generated or similar things that would interfere with the CA rotation sequence.
```

File: docs/proposals/18-shoot-CA-rotation.md [hunk 9]
```diff
@@ -71,30 +71,30 @@ In each of those cases we currently need to basically manually recreate and repl
...
-  - `gardener-apiserver` performs the needful validation: user can't trigger another rotation if one is already in progress, user can't trigger `complete-ca-rotation` if first phase has not been compeleted, etc.
-- The annotation triggers a usual shoot reconciliation (just like a kubeconfig or SSH key rotation)
-- gardenlet begins the CA rotation sequence by setting the new status section `.status.credentials.caRotation` (probably in `updateShootStatusOperationStart`) and removes the annotation afterwards
-  - shoot reconciliation needs to be idemptotent to CA rotation phase, i.e. if a usual reconciliation or maintenance operation is triggered in between, no new CAs are generated or similar things that would interfere with the CA rotation sequence
+  - Annotation increases the generation
+  - Value for triggering first phase: `start-ca-rotation`
+  - Value for triggering the second phase: `complete-ca-rotation`
+  - `gardener-apiserver` performs the needful validation: a user can't trigger another rotation if one is already in progress, a user can't trigger `complete-ca-rotation` if first phase has not been compeleted, etc.
```
Comment (acumino) [inline docs/proposals/18-shoot-CA-rotation.md]:
```suggestion
  - `gardener-apiserver` performs the needful validation: a user can't trigger another rotation if one is already in progress, a user can't trigger `complete-ca-rotation` if first phase has not been completed, etc.
```

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by ialidzhikov — COMMENTED** [review]
/approve

**Pull request merged**